feature_engine/selection/target_mean_encoding.py

```diff
@@ -0,0 +1,264 @@
+        --------------
+
+            Public methods
+            --------------
+                `fit(self, X, y)`
+                `transform(self)`
+                `fit_transform(self, X, y)`
+        
+    Parameters
+    ----------
+
+    variables: list, default=None
+        The list of variables to evaluate. If None, the transformer will evaluate all 
+        variables in the dataset associated with the variables_type.
+
+    scoring: string, default='roc_auc_score'
+        This indicates the metrics score to perform the feature selection.
+        The current support includes 'roc_auc_score' and 'r2_score'.
+
+
+    """
+
+    def __init__(self, variables=None, scoring='roc_auc_score'):
+        self.variables = _define_variables(variables)
+        self.scoring = scoring
+        self.X_test_enc = None
+        self.y_test = None
+
+    def fit(self, X, y, quantiles=5, test_size=0.3, random_state=0):
+        """
+        Performs
+        --------
+
+            1. Handling missing values and check invalid input variables.
+            2. Split data into train and test.
+            3. Performs the target mean encoding.
+            4. Update the results X_train_enc and X_test_enc.
+        
+        Returns
+        -------
+            None
+
+        Parameters
+        ----------
+
+            X: pandas dataframe of shape = [n_samples, n_features]
+                The input dataframe.
+
+            y: None
+                y is not needed for this transformer. You can pass y or None.
+            
+            test_size: float, default=0.3
+                The test size setting of the data in the train_test_split method.
+
+            random_state: int, default=0
+                The random state setting in the train_test_split method.
+
+            quantiles: int, default=5
+                The amount of quantiles setting in the qcut of the fit method in case of the numerical types.
+                The range of the quantiles values is in between 3 and 100. The transformer will return a message
+                if the input quantiles value is out of this range.
+
+        Attributes
+        ----------
+
+            X_test_enc: pandas dataframe of shape = [n_samples, n_features]
+                The output of target mean encoding of the test set.
+        
+        """
+
+        # check the target
+        if y is None:
+            raise ValueError('Please provide a target y for this encoding method.')
+
+        # check input dataframe
+        X = _is_dataframe(X)
+
+        # find all variables or check those entered are in the dataframe
+        self.variables = _find_all_variables(X, self.variables)
+        
+        # find categorical variables or check that those vars entered by the user
+        # are of type object
+        cat_variables = _find_categorical_variables(X[self.variables] if self.variables else X)
+        num_variables = _find_numerical_variables(X[self.variables] if self.variables else X)
```

Comment:
I think it might be better to rename these as:

```
variables_categorical
variables_numerical
```

and we need to make them part of self.

so:

```
self.variables_categorical
self.variables_numerical
```